app/results.py

```diff
@@ -120,22 +120,27 @@ def reject_query(self, query):
             '/query', role=self.params.role, error='error', query=query.query)
 
     def get_results_url(self, query_name, query_location):
-        return self.get_url(
-            '/results',
-            ui='' if self.env.ui == 'small' else self.env.ui,
-            query_name=query_name,
-            query_location=query_location,
-            role=self.params.role,
-            given_name=self.params.given_name,
-            family_name=self.params.family_name)
+        params = {
+            'query_name': query_name,
+            'query_location': query_location,
+            'role': self.params.role,
+            'given_name': self.params.given_name,
+            'family_name': self.params.family_name,
+        }
+        if self.env.ui == 'small':
+            params['ui'] = None  # Links to the default UI.
```

Comment:
get_url() preserves subset of URL parameters (including "ui") from the current URL by default. So it still preserves "ui" param unless it's "small". Updated the comment to clarify it.

It *removes* the param from the URL if None is explicitly passed. It's a bit confusing, though.